src/lib/builders/color-picker/create.ts

```diff
@@ -0,0 +1,898 @@
+import { addMeltEventListener, builder, createElHelpers, executeCallbacks, isBrowser } from "$lib/internal/helpers";
+import type { Defaults } from "$lib/internal/types";
+import { onMount } from "svelte";
+
+import type { ArrowKeys, ColorHSL, ColorHSV, ColorPickerParts, ColorRGB, CreateColorPickerProps, EyeDropperType, EyeDropperWindow, KeyDurations, NodeElement, NodeSize, Position, ReturnedColor } from "./types";
+import { derived, get, writable, type Readable, type Writable } from "svelte/store";
+
+
+const defaults = {
+    forceVisible: false,
+    defaultColor: '#ff0000',
+    hueSliderOrientation: 'horizontal',
+    alphaSliderOrientation: 'horizontal',
+} satisfies Defaults<CreateColorPickerProps>;
+
+const { name } = createElHelpers<ColorPickerParts>('color-picker');
+
+
+export function createColorPicker(args?: CreateColorPickerProps) {
+    const argsWithDefaults = { ...defaults, ...args };
+
+    let dragging = false;
+    let hueDragging = false;
+    let alphaDragging = false;
+    let insideUpdate = false;
+    let inputUpdate = false;
+    const speepUpStep = 5;
```

Comment:
What does this variable do?

Reply:
@TGlide that's a typo actually, didn't notice that. It's supposed to be `speedUpStep`, basically when you use arrow keys to move the color picker around the main canvas, the picker moves pretty slowly (1 pixel at a time), so I added this larger step size that gets activated after an arrow key has been pressed for more than 1 second.